commands/command_test.go

```diff
@@ -0,0 +1,74 @@
+package commands
+
+import (
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/mattermost/mmctl/printer"
+	"github.com/spf13/cobra"
+)
+
+func (s *MmctlUnitTestSuite) TestCreateCommandCmdF() {
+	s.Run("Create a new custom slash command for a specified team", func() {
+		printer.Clean()
+		teamArg := "example-team-id"
+		titleArg := "example-command-name"
+		descriptionArg := "example-description-text"
+		triggerWordArg := "example-trigger-word"
+		urlArg := "http://localhost:8000/example"
+		creatorUsernameArg := "example-username"
+		responseUsernameArg := "example-username2"
+		iconArg := "icon-url"
+		method := "G"
+		autocomplete := false
+		autocompleteDesc := "autocompleteDesc"
+		autocompleteHint := "autocompleteHint"
+
+		mockTeam := model.Team{Id: teamArg}
+		mockCreator := model.User{Username: creatorUsernameArg}
+		mockCommand := model.Command{
+			CreatorId:        creatorUsernameArg,
+			TeamId:           teamArg,
+			Trigger:          triggerWordArg,
+			Username:         responseUsernameArg,
+			IconURL:          iconArg,
+			DisplayName:      titleArg,
+			Description:      descriptionArg,
+			URL:              urlArg,
+			Method:           method,
+			AutoComplete:     autocomplete,
+			AutoCompleteDesc: autocompleteDesc,
+			AutoCompleteHint: autocompleteHint,
+		}
+
+		cmd := &cobra.Command{}
+		cmd.Flags().String("team", teamArg, "")
+		cmd.Flags().String("title", titleArg, "")
+		cmd.Flags().String("description", descriptionArg, "")
+		cmd.Flags().String("trigger-word", triggerWordArg, "")
+		cmd.Flags().String("url", urlArg, "")
+		cmd.Flags().String("creator", creatorUsernameArg, "")
+		cmd.Flags().String("response-username", responseUsernameArg, "")
+		cmd.Flags().String("icon", iconArg, "")
+
+		// createCommandCmdF will call getTeamFromTeamArg,  getUserFromUserArg,
+		s.client.
+			EXPECT().
+			GetTeam(teamArg, "").
+			Return(&mockTeam, &model.Response{Error: nil}).
+			Times(1)
+		s.client.
+			EXPECT().
+			GetUserByEmail(creatorUsernameArg, "").
+			Return(&mockCreator, &model.Response{Error: nil}).
+			Times(1)
+		s.client.
+			EXPECT().
+			CreateCommand(cmd).		// ERROR ; passing in the wrong argument???
```

Comment:
not sure why its the wrong argument

Reply:
you are passing the `cobra.Command` instance instead of the `model.Command` that you've just created. Try with `mockCommand` instead

Reply:
fixed that part but i am getting this error right now.  

![Screen Shot 2019-11-17 at 12 24 05 PM](https://user-images.githubusercontent.com/7275313/69013484-5c9e0d00-0935-11ea-9a77-cc1e397f0f50.png)

Am i doing line 48 wrong?